src/API/Site/Controllers/Assets/AssetSuggestionsController.php

```diff
@@ -0,0 +1,162 @@
+	 * @return array
+	 */
+	public function get_collection_params(): array {
+		return [
+			'search'   => [
+				'description'       => __( 'Search for post title or term name', 'google-listings-and-ads' ),
+				'type'              => 'string',
+				'default'           => '',
+				'sanitize_callback' => 'sanitize_text_field',
+				'validate_callback' => 'rest_validate_request_arg',
+			],
+			'per_page' => [
+				'description'       => __( 'The number of items to be return', 'google-listings-and-ads' ),
+				'type'              => 'number',
+				'default'           => 30,
+				'sanitize_callback' => 'absint',
+				'minimum'           => 1,
+				'validate_callback' => 'rest_validate_request_arg',
+			],
+			'order_by' => [
+				'description'       => __( 'Sort retrieved items by parameter', 'google-listings-and-ads' ),
+				'type'              => 'string',
+				'default'           => 'title',
+				'sanitize_callback' => 'sanitize_text_field',
+				'enum'              => [ 'id', 'type', 'title', 'url' ],
+				'validate_callback' => 'rest_validate_request_arg',
+			],
+		];
+	}
+
+	/**
+	 * Get the callback function for the list of final-url suggestions request.
+	 *
+	 * @return callable
+	 */
+	protected function get_final_urls_suggestions_callback(): callable {
+		return function( Request $request ) {
+			$search   = $request->get_param( 'search' );
+			$per_page = $request->get_param( 'per_page' );
+			$order_by = $request->get_param( 'order_by' );
+			return array_map(
+				function( $item ) use ( $request ) {
+					$data = $this->prepare_item_for_response( $item, $request );
+					return $this->prepare_response_for_collection( $data );
+				},
+				$this->asset_suggestions_service->get_final_urls_suggestions( $search, $per_page, $order_by )
+			);
+		};
+	}
+
+
+
+	/**
+	 * Get the item schema for the controller.
+	 *
+	 * @return array
+	 */
+	protected function get_schema_properties(): array {
+		return [
+			'id'    => [
+				'type'        => 'number',
+				'description' => __( 'Post ID or Term ID', 'google-listings-and-ads' ),
+				'context'     => [ 'view' ],
+			],
+			'type'  => [
+				'type'        => 'string',
+				'description' => __( 'Post or term', 'google-listings-and-ads' ),
+				'context'     => [ 'view' ],
+				'enum'        => [ 'post', 'term' ],
+				'readonly'    => true,
+			],
+			'title' => [
+				'type'        => 'string',
+				'description' => __( 'The post or term title', 'google-listings-and-ads' ),
+				'context'     => [ 'view' ],
+				'readonly'    => true,
+			],
+			'url'   => [
+				'type'        => 'string',
+				'description' => __( 'The URL linked to the post/term', 'google-listings-and-ads' ),
+				'context'     => [ 'view' ],
+				'readonly'    => true,
+			],
+
+		];
+
+	}
+
+
+	/**
+	 * Get the item schema name for the controller.
+	 *
+	 * Used for building the API response schema.
+	 *
+	 * @return string
+	 */
+	protected function get_schema_title(): string {
+		return 'asset_final_urls_suggestions';
```

Comment:
💅  Minor

Same as in https://github.com/woocommerce/google-listings-and-ads/pull/1784/files#r1031025484

Plural/singular inconsistencies